log2ram

```diff
@@ -70,25 +70,34 @@ case "$1" in
       mount --bind $RAM_LOG/ $HDD_LOG/
       mount --make-private $HDD_LOG/
       wait_for $HDD_LOG
-	if [ "$ZL2R" = true ]; then
+	  if [ "$ZL2R" = true ]; then
 		createZramLogDrive
-		mount -t ext4 -o nosuid,noexec,nodev,user=log2ram /dev/zram${RAM_DEV} ${RAM_LOG}/
-	else
+		mount -t ext4 -o nosuid,noexec,nodev,discard,user=log2ram /dev/zram${RAM_DEV} ${RAM_LOG}/
+	  else
 		mount -t tmpfs -o nosuid,noexec,nodev,mode=0755,size=${SIZE} log2ram $RAM_LOG/
-	fi
+	  fi
       wait_for $RAM_LOG
       syncFromDisk
       ;;
 
   stop)
       syncToDisk
-	#ZRAM_LOG=$(awk '$2 == "/var/log" {print $1}' /proc/mounts)
-	#ZRAM_LOG=$(echo ${ZRAM_LOG} | grep -o -E '[0-9]+')
+	  invoke-rc.d rsyslog stop
```

Comment:
I'm not sure about stopping rsyslog.
You will lose all pending log this way.
The previous way was to do nothing, so, the worst case was : some log aren't copy to the HDD. In you example, the occurrence is higher because a restart of a daemon can take time.

Reply:
No you don't lose all pending log as it was an example.
rsyslog starts on a socket so any output restarts rsyslog that is why its such a pain to stop you get a small race but also you can start that way without the need for such an early service if needed and restart rsyslog
It why you have lazy umount as a straight umount would fail as rsyslog have open files.
You can stop rsyslog at any time its done on each logrotate and hence at times you will see the HUP message.

But again its an example, change and use as needed and maybe actually have an install of your program so you can test and debug?

Reply:
Do you want a pi zero ? email me stuartiannaylor@outlook.com

Reply:
Azlux I deleted my log2zram repo reinitialised a straight branch to yours and spent the time just to add those bits and also just show a few others.
Git is a great tool for some quick and dirty collaboration where its fast and simple to share each others branches.
Its so frustrating to actually be running / testing and providing to the owner who can not even be bothered to run an instance or just quickly clone a suggestion and just give it a try.

We are not Linus Torvalds and this is not the Linux Kernel its just an extremely dirty script to allow some IoT, Maker tinkers help with SD wear.

I will be honest and IMHO you and Log2Ram is a total cul-de-sac and another bad script that because of link presence is often emulated copied and used because a load of noobs like myself believe that link presence must mean it is indicative of quality.

I am only posting the relative info and I am refusing to do 100% of your work because it is a cul-de-sac primarily due to you.
If you have suggestions from others run up a branch and invite collaboration and give users a choice, it is that simple.
I am reading in between the lines of your replies and it has been quite a while now that personally feel its a doomed cul-de-sac by a non co-operative author who does even have a instance or is prepared to test any suggestions whilst making some extremely strange technical comments and decisions.
Its such a complete waste of my time that whilst being the owner of a project you can not even be bothered to run an instance yourself.

Its why I decided a week ago that deliberately I will fork and get away as adding to the link presence is just misguiding many like myself of lower competence to waste much time with a bad script.